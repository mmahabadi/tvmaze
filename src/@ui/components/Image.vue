<script setup lang="ts">
import { defineProps } from 'vue'

defineProps<{ src?: string; alt?: string }>()
</script>
<template>
  <img v-if="src" :src="src" :alt="alt" class="w-full h-full object-cover" />
  <img
    v-else
    class="w-full h-full object-cover"
    src="@/assets/image-placeholder.jpg"
    alt="image-placeholder"
  />
</template>
