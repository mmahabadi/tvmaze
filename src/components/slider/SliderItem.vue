<script setup lang="ts">
import { computed } from 'vue'
import DateIcon from '@/@ui/components/icons/CalendarIcon.vue'
import ClockIcon from '@/@ui/components/icons/ClockIcon.vue'
import IconPlay from '@/components/icons/IconPlay.vue'

const { name, image, endDate, rating, genre } = defineProps<{
  name: string
  image: string
  endDate: string
  rating: number
  genre: string
}>()

const year = computed(() => new Date(endDate).getFullYear())
</script>
<template>
  <img :src="image" alt="poster" class="w-full h-full object-cover object-top" />
  <div
    class="bg-black/40 absolute top-0 bottom-0 right-0 left-0 flex flex-col justify-center items-center"
  >
    <div
      class="details transition-opacity duration-200 ease-in opacity-0 flex flex-col justify-center md:gap-5 gap-4 items-center"
    >
      <h1 class="xl:text-4xl truncate capitalize font-sans sm:text-2xl text-4xl font-bold">
        {{ name }}
      </h1>
      <div class="flex gap-5 items-center">
        <div class="flex items-center gap-2">
          <span class="text-sm font-medium">{{ genre }}</span>
        </div>
        <div class="flex items-center gap-2">
          <DateIcon class="text-teal-500" /><span class="text-sm font-medium">{{ year }}</span>
        </div>
        <div class="flex items-center gap-2">
          <ClockIcon class="text-teal-500" /><span class="text-sm font-medium">{{ rating }}hr</span>
        </div>
      </div>
      <div class="flex gap-5 items-center">
        <a
          class="watch-button bg-teal-500 hover:bg-teal-700 text-white font-bold py-2 px-4 rounded inline-flex items-center"
          :href="name"
        >
          <IconPlay class="text-white mr-2" />
          <IconPlay class="text-white mr-2 opacity-0 absolute animate-ping" />
          Watch
        </a>
      </div>
    </div>
  </div>
</template>
<style scoped>
.swiper-slide-next .details,
.watch-button:hover .animate-ping {
  opacity: 100;
}
</style>
