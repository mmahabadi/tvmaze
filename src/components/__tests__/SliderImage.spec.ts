import { describe, it, expect, beforeEach } from 'vitest'
import { mount, RouterLinkStub, VueWrapper } from '@vue/test-utils'
import SliderItem from '../slider/SliderItem.vue'
import DateIcon from '@/@ui/components/icons/CalendarIcon.vue'
import ClockIcon from '@/@ui/components/icons/ClockIcon.vue'
import IconPlay from '@/@ui/components/icons/IconPlay.vue'
import Image from '@/@ui/components/Image.vue'

describe('SliderImage', () => {
  let wrapper: VueWrapper<unknown>
  const props = {
    id: 1,
    name: 'The Simpsons',
    image: 'The Simpsons.jpg',
    endDate: '1989',
    rating: 8.7,
    genre: 'Animation'
  }

  beforeEach(() => {
    wrapper = mount(SliderItem, {
      props,
      global: {
        stubs: {
          DateIcon,
          ClockIcon,
          IconPlay,
          Image,
          'router-link': RouterLinkStub
        }
      }
    })
  })

  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
