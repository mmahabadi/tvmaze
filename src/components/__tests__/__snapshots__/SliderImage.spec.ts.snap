// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SliderImage > renders correctly 1`] = `
"<img data-v-1e279afe="" src="The Simpsons.jpg" alt="poster" class="w-full h-full object-cover poster">
<div data-v-1e279afe="" class="bg-overlay">
  <div data-v-1e279afe="" class="details">
    <h1 data-v-1e279afe="" class="text-heading">The Simpsons</h1>
    <div data-v-1e279afe="" class="info-section">
      <div data-v-1e279afe="" class="info-part"><span data-v-1e279afe="" class="info-text">Animation</span></div>
      <div data-v-1e279afe="" class="info-part"><svg data-v-1e279afe="" stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 448 512" class="text-subMain w-3 h-3 text-teal-500" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg">
          <path d="M148 288h-40c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12zm108-12v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm96 0v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm-96 96v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm-96 0v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm192 0v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm96-260v352c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V112c0-26.5 21.5-48 48-48h48V12c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v52h128V12c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v52h48c26.5 0 48 21.5 48 48zm-48 346V160H48v298c0 3.3 2.7 6 6 6h340c3.3 0 6-2.7 6-6z"></path>
        </svg><span data-v-1e279afe="" class="info-text">1989</span></div>
      <div data-v-1e279afe="" class="info-part"><svg data-v-1e279afe="" stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 24 24" class="text-subMain w-3 h-3 text-teal-500" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg">
          <path d="M12 2C6.486 2 2 6.486 2 12s4.486 10 10 10 10-4.486 10-10S17.514 2 12 2zm0 18c-4.411 0-8-3.589-8-8s3.589-8 8-8 8 3.589 8 8-3.589 8-8 8z"></path>
          <path d="M13 7h-2v6h6v-2h-4z"></path>
        </svg><span data-v-1e279afe="" class="info-text">8.7hr</span></div>
    </div>
    <div data-v-1e279afe="" class="info-section"><a data-v-1e279afe="" class="watch-button"><svg data-v-1e279afe="" class="svg-icon text-white mr-2" style="width: 1em; height: 1em; vertical-align: middle; fill: currentColor; overflow: hidden;" viewBox="0 0 1024 1024" version="1.1" xmlns="http://www.w3.org/2000/svg">
          <path d="M852.727563 392.447107C956.997809 458.473635 956.941389 565.559517 852.727563 631.55032L281.888889 993.019655C177.618644 1059.046186 93.090909 1016.054114 93.090909 897.137364L93.090909 126.860063C93.090909 7.879206 177.675064-35.013033 281.888889 30.977769L852.727563 392.447107 852.727563 392.447107Z"></path>
        </svg><svg data-v-1e279afe="" class="svg-icon text-white mr-2 opacity-0 absolute animate-ping" style="width: 1em; height: 1em; vertical-align: middle; fill: currentColor; overflow: hidden;" viewBox="0 0 1024 1024" version="1.1" xmlns="http://www.w3.org/2000/svg">
          <path d="M852.727563 392.447107C956.997809 458.473635 956.941389 565.559517 852.727563 631.55032L281.888889 993.019655C177.618644 1059.046186 93.090909 1016.054114 93.090909 897.137364L93.090909 126.860063C93.090909 7.879206 177.675064-35.013033 281.888889 30.977769L852.727563 392.447107 852.727563 392.447107Z"></path>
        </svg> Watch </a></div>
  </div>
</div>"
`;
